lib/async.js

```diff
@@ -834,8 +834,21 @@
                 if (q.tasks.length === q.concurrency) {
                     q.saturated();
                 }
-                async.setImmediate(q.process);
             });
+            setImmediate(q.process);
```

Comment:
Use async.setImmediate here, so it falls back to other methods when it's not available in the environment.